system/posix/retcode/retcode.go

```diff
@@ -0,0 +1,27 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package retcode
+
+import "hash/crc32"
+
+// Calc will calculate an POSIX retcode from an error.
```

Comment:
I'm not sure why we want to do this -- it looks like we're collapsing all errors into a range of code 1-255.

https://unix.stackexchange.com/a/418802 suggests that the range 1-125 might be more reasonable, or 1 - (2^31-1) if you're willing to work with `waitid` and some of the newer posix calls.